otx/algorithms/action/utils/convert_public_data_to_cvat.py

```diff
@@ -0,0 +1,260 @@
+
+    phases = ["train", "valid"]
+    for phase in phases:
+        csv_path = osp.join(src_path, f"annotations/{phase}.csv")
+        with open(csv_path, "r", encoding="utf-8") as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter=",")
+            pathlib.Path(osp.join(dst_path, phase)).mkdir(parents=True, exist_ok=True)
+
+            video_dict = {}
+            for i, line in enumerate(csv_reader):
+                video_id, frame_idx, bboxes, class_idx = line[0], line[1], line[2:6], line[6]
+                video_path = osp.join(frames_dir_path, video_id)
+                frame_list = os.listdir(video_path)
+
+                if video_id not in video_dict:
+                    annotations, _ = generate_default_cvat_xml_fields(i, class_idx, video_path, frame_list)
+                    track = etree.Element("track", id=str(i), label=str(class_idx))
+                    box = etree.Element(
+                        "box",
+                        frame=str(frame_idx),
+                        outside="0",  # not used
+                        occluded="0",  # not used
+                        xtl=str(bboxes[0]),
+                        ytl=str(bboxes[1]),
+                        xbr=str(bboxes[2]),
+                        ybr=str(bboxes[3]),
+                        z_order="0",
+                    )
+                    box.text = ""
+                    track.append(box)
+                    annotations.append(track)
+                    video_dict[video_id] = annotations
+                else:
+                    track = etree.Element("track", id=str(i), label=str(class_idx))
+                    box = etree.Element(
+                        "box",
+                        frame=str(frame_idx),
+                        outside="0",  # not used
+                        occluded="0",  # not used
+                        xtl=str(bboxes[0]),
+                        ytl=str(bboxes[1]),
+                        xbr=str(bboxes[2]),
+                        ybr=str(bboxes[3]),
+                        z_order="0",
+                    )
+                    box.text = ""
+                    track.append(box)
+                    video_dict[video_id].append(track)
+
+            for i, (_, v) in enumerate(video_dict.items()):
+                shutil.copytree(video_path, osp.join(dst_path, phase, f"video_{i}/images"))
+                et = etree.ElementTree(v)
+                et.write(
+                    osp.join(dst_path, phase, f"video_{i}/annotations.xml"),
+                    pretty_print=True,
+                    xml_declaration=True,
+                    encoding="utf-8",
+                )
+
+
+def main(src_path, dst_path):
+    """Main function."""
+    # convert_jester_dataset_to_datumaro(src_path, dst_path)
+    convert_ava_dataset_to_datumaro(src_path, dst_path)
+
+
+if __name__ == "__main__":
+    main("/local/sungmanc/datasets/ava_SC", "/local/sungmanc/datasets/ava_SC_cvat_multifolder_detection")
```

Comment:
Is this supposed to be deleted?

Reply:
Ah, yeah it will be changed to take the user's input